File: python/generating/factory/MainFactory.py
from enum import Enum
from abc import ABC, abstractmethod

class CarType(Enum):
    PassengerCar = "Passenger car"
    SUV = "SUV"

class Car(ABC):
    @abstractmethod
    def getCarType(self):
        pass

class PassengerCar(Car):
    def getCarType(self):
        return CarType.PassengerCar.value

class SUV(Car):
    def getCarType(self):
        return CarType.SUV.value

class CarFactory:
    def getCar(self, type):
        if (isinstance(type, CarType)):
            match type:
                case CarType.PassengerCar:
                    return PassengerCar()
                case CarType.SUV:
                    return SUV()
        else:
            return None

def main():
    carFactory = CarFactory()
    car = carFactory.getCar(CarType.PassengerCar)
    print(car.getCarType())
    car = carFactory.getCar(CarType.SUV)
    print(car.getCarType())

main()

File: swift5/Sources/swift5/generating/factory/MainFactory.swift

fileprivate enum CarType: String {
    case PassengerCar = "Passenger car"
    case SUV = "SUV"
}

fileprivate protocol Car {
    func getCarType() -> String;
}

fileprivate class PassengerCar : Car {
	public func getCarType() -> String {
		return CarType.PassengerCar.rawValue;
	}
}

fileprivate class SUV : Car {
	public func getCarType() -> String {
		return CarType.SUV.rawValue;
	}
}

fileprivate class CarFactory {
    public func getCar(value: CarType) -> Car? {
        switch value {
        case .PassengerCar:
			return PassengerCar();
		case .SUV:
			return SUV();
		default:
			return nil;
		} 
    }
}

public class MainFactory {
	public static func main() {
		let carFactory: CarFactory = CarFactory();
		var car: Car? = carFactory.getCar(value: CarType.PassengerCar);
		print(car!.getCarType());
		car = carFactory.getCar(value: CarType.SUV);
		print(car!.getCarType());
	}
}